### PR TITLE
fix(core): Don't unpin anchor proof, merkle tree, or CACAO when unpinning streams

### PR DESCRIPTION
**Pull request by stbrody**
This does technically leave some leaked CIDs pinned in IPFS after unpinning a stream, but that seems better to me than accidentally unpinning data that might be in use by other streams.

This PR also dedupes some code and improves the test coverage of the pin store

### CODE REVIEW

File: packages/core/src/store/__tests__/pin-store.test.ts [hunk 1]
```diff
@@ -229,13 +273,96 @@ describe('#add', () => {
...
+    expect(pinning.unpin).toBeCalledTimes(4)
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    expect(pinning.unpin.mock.calls[0][0].toString()).toEqual(
+      stateWithSignedCommit.log[0].cid.toString()
+    )
+    expect(pinning.unpin.mock.calls[1][0].toString()).toEqual(
+      stateWithSignedCommit.log[1].cid.toString()
+    )
+    expect(pinning.unpin.mock.calls[2][0].toString()).toEqual(capCID.toString())
```
Comment (ukstv) [inline packages/core/src/store/__tests__/pin-store.test.ts:322]: So, we unpin a capability, even if it can be used by some other stream?

**Review by ukstv — CHANGES_REQUESTED** [review]
LGTM for the change in the description.

Not quite relevant: Do you think we can leave a capability pinned? There is no guarantee a capability is issued just for a particular stream. Most often it is reused across many streams.

**Comment by stbrody** [comment]
> Do you think we can leave a capability pinned? There is no guarantee a capability is issued just for a particular stream. Most often it is reused across many streams.

Ooh really good catch.  Yeah I'll update this to leave the CACAO alone during unpin, just like I do with the anchor proof and merkle tree

**Review by ukstv — APPROVED** [review]
🏇

**Pull request merged**